source/logging/logging.rst

```diff
@@ -0,0 +1,416 @@
+facilitate support of drivers for our internal teams, and improve our documentation around
+troubleshooting.
+
+Design Rationale
+================
+
+-----------------------------
+Truncation of large documents
+-----------------------------
+
+1. Why have an option? 
+    We considered a number of approaches for dealing with documents of potentially very large size
+    in log messages, e.g. command documents, including 1) always logging the full document, 2) only
+    logging documents with the potential to be large when the user opts in, and 3) truncating large
+    documents by default, but allowing the user to adjust the maximum length logged. We chose the
+    third option as we felt it struck the best balance between concerns around readability and
+    usability of log messages. In the case where data is sufficiently small, the default behavior
+    will show the user the full data. In the case where data is large, the user will receive a
+    readable message with truncated data, but have the option to see more or all of the data.
+
+2. Why are the units for max document length flexible? 
+    String APIs vary across languages, and not all drivers will be able to easily and efficiently
+    truncate strings in the same exact manner. The important thing is that the option exists and
+    that its default value is reasonable, and for all possible unit choices (byte, code point, 
+    code unit, or grapheme) we felt 1000 was a reasonable default. See
+    `here <https://exploringjs.com/impatient-js/ch_unicode.html>`_ for a helpful primer on related
+    Unicode concepts.
+
+--------------------------------------
+Structured versus Unstructured Logging
+--------------------------------------
+The MongoDB server produces structured logs as of 4.4, so it seems natural that MongoDB drivers
+might too. However, the idiomatic logging mechanisms of choice for some language ecosystems (e.g.
+Java) do not support structured logging yet, so we cannot require it.
+
+--------------------------------------
+Omitting Null Values from Log Messages
+--------------------------------------
+We considered alternatives such as allowing, or requiring, drivers to explicitly include null values
+in log messages. While this might make it easier to identify cases where a value is unexpectedly null,
+we decided against it because there are a number of values that will often be null, or even always
+be null for certain applications (e.g. ``serviceId`` when not connected to a load-balanced topology)
+and their inclusion may confuse users and lead them to think the null value is meaningful.
+Additionally, always including null values would increase the size of log messages.
+
+Backwards Compatibility
+=======================
+This specification takes the stance that the contents of log messages (both structured and
+unstructured) are _not_ covered by semantic versioning, but that logging components _are_, since
```

Comment:
Are _not_/_are_ missing additional underscore?

Reply:
I think it should actually be asterisks instead of underscores, I meant to make it italic. fixed! thanks